### PR TITLE
Native support of Promises for data loading

### PR DESCRIPTION
**Pull request by Elhebert**
I added the native support of promises. use infinite-scroll-[top-]promise="true" to enable the promises.

For the $http.get example on the website, you could now use this, and no longer use the busy variable to make the system wait for the promise to be resolved, it's now done directly in the directive.

### CODE REVIEW

File: src/infinite-scroll.coffee [hunk 1]
```diff
@@ -146,16 +178,29 @@ mod.directive 'infiniteScroll', ['$rootScope', '$window', '$interval', 'THROTTLE
     scope.$watch 'infiniteScrollUseDocumentBottom', handleInfiniteScrollUseDocumentBottom
     handleInfiniteScrollUseDocumentBottom scope.infiniteScrollUseDocumentBottom
 
+    # allow to use promises to load more data.
+    # This option will wait for the promise to be resolve before permitting
```
Comment (graingert) [inline src/infinite-scroll.coffee]: "This option will wait for the promise to resolve before awaiting a new one"

or, "This option will await each promise before awaiting the next promise."

Or "When true, ngInfiniteScroll will await each promise serially"

Comment (Elhebert) [inline src/infinite-scroll.coffee]: :+1: Those are better formulation.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,5 @@
+root = true
```
Comment (graingert) [inline .editorconfig]: please remove this file.

Comment (Elhebert) [inline .editorconfig]: Is there a good reason ? 

Just wondering, 'cause I find the `.editorconfig` quite useful.

Comment (graingert) [inline .editorconfig]: it could be considered in another PR.
